File: Editorial/2001-2100/2078_Two_Furthest_Houses_With_Different_Colors/Python3/2_Greedy.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        ans = 0
        n = len(colors)
        for i in range(n - 1):
            if colors[i] != colors[n - 1]:
                ans = max(ans, i, n - 1 - i)
        return ans

File: Editorial/2001-2100/2078_Two_Furthest_Houses_With_Different_Colors/Java/2_Greedy.java
class Solution {
    public int maxDistance(int[] colors) {
        int n = colors.length;
        int ans = 0;
        for (int i = 0; i < n - 1; i++) {
            if (colors[i] != colors[n - 1]) {
                ans = Math.max(ans, Math.max(i, n - 1 - i));
            }
        }
        return ans;
    }
}
